File: leetcode/Algorithms/718.MaximumLengthOfRepeatedSubarray/Solution.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int result = 0, aStart = A.length - 1, bStart = 0;
        while (bStart < B.length) {
            int tmp = 0;
            if (aStart > 0) {
                for (int i = 0; i < Math.min(B.length, A.length - aStart); i++) {
                    if (B[i] == A[i + aStart]) {
                        tmp++;
                        result = Math.max(tmp, result);
                    } else {
                        tmp = 0;
                    }
                }
                aStart--;
            } else {
                for (int i = bStart; i < Math.min(B.length, bStart + A.length); i++) {
                    if (B[i] == A[i - bStart]) {
                        tmp++;
                        result = Math.max(tmp, result);
                    } else {
                        tmp = 0;
                    }
                }
                bStart++;
            }
        }
        return result;
    }
}

File: leetcode/Algorithms/718.MaximumLengthOfRepeatedSubarray/Solution.py
class Solution:
    def findLength(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: int
        """
        matrix = [[0 for _ in range(len(B) + 1)] for _ in range((len(A) + 1))]
        for i in range(0, len(A)):
            for j in range(0, len(B)):
                if A[i] == B[j]:
                    matrix[i + 1][j + 1] = 1 + matrix[i][j]
        return max(max(row) for row in matrix)

